src/test/groovy/com/intuit/graphql/orchestrator/integration/NestedFieldResolverSpec.groovy

```diff
@@ -0,0 +1,133 @@
+            (QUERY_B) : [ data: [
+                    bTopField: [[ bObjectField: "bObjectFieldValue1"], [ bObjectField: "bObjectFieldValue2"]]
+            ]]
+    ]
+
+    def schemaC = """
+        type Query {
+            cTopField: CObjectType!
+        }
+
+        type CObjectType {
+            cObjectField: String
+        }
+    """
+
+    def QUERY_C = "query Resolver_Directive_Query {cTopField_0:cTopField {cObjectField} cTopField_1:cTopField {cObjectField}}"
+    def mockServiceResponseC = [
+            (QUERY_C): [data: [
+                    cTopField_0: [ cObjectField: "cObjectFieldValue1"],
+                    cTopField_1: [ cObjectField: "cObjectFieldValue2"],
+            ]]
+    ]
+
+    GraphQLSchema graphQLSchema
+
+    @Subject
+    GraphQLOrchestrator specUnderTest
+
+    void setup() {
+        def serviceA = createQueryMatchingService("ServiceA", schemaA, mockServiceResponseA)
+        def serviceB = createQueryMatchingService("ServiceB", schemaB, mockServiceResponseB)
+        def serviceC = createQueryMatchingService("ServiceC", schemaC, mockServiceResponseC)
+        specUnderTest = createGraphQLOrchestrator([serviceA, serviceB, serviceC])
+        graphQLSchema = specUnderTest.getSchema()
+    }
+
+    def "Nested FieldResolver"() {
+        given:
+        def graphqlQuery =    """
+            {
+                bTopField {
+                    bObjectField
+                    aTopField {
+                        aObjectField  
+                        cTopField {
+                            cObjectField
+                        }                                              
+                    }
+                }
+            }
+        """
+        ExecutionInput executionInput = createExecutionInput(graphqlQuery)
+
+        when:
+        ExecutionResult executionResult = specUnderTest.execute(executionInput).get()
+
+        then:
+        executionResult.getErrors().isEmpty()
+        executionResult?.data?.bTopField?.size() == 2
+        executionResult?.data?.bTopField[0]?.size() == 2
+        executionResult?.data?.bTopField[1]?.size() == 2
+        executionResult?.data?.bTopField[0]?.bObjectField == "bObjectFieldValue1"
+        executionResult?.data?.bTopField[0]?.aTopField?.size() == 2
+        executionResult?.data?.bTopField[0]?.aTopField?.aObjectField == "aObjectFieldValue1"
+        executionResult?.data?.bTopField[0]?.aTopField?.cTopField?.size() == 1
+        executionResult?.data?.bTopField[0]?.aTopField?.cTopField?.cObjectField == "cObjectFieldValue1"
+        executionResult?.data?.bTopField[1]?.bObjectField == "bObjectFieldValue2"
+        executionResult?.data?.bTopField[1]?.aTopField?.size() == 2
+        executionResult?.data?.bTopField[1]?.aTopField?.aObjectField == "aObjectFieldValue2"
+        executionResult?.data?.bTopField[1]?.aTopField?.cTopField?.size() == 1
+        executionResult?.data?.bTopField[1]?.aTopField?.cTopField?.cObjectField == "cObjectFieldValue2"
+    }
```

Comment:
can we assert on the outgoing queries to services A,B & C

Reply:
The MockServiceProvider used is not using Mockito.  What it does is it returns a data based on test query - response Map.  If the query does not match, it throws an exception and the test fails.